2020-01-Winter/2_string_manipulation/README.md

```diff
@@ -94,21 +99,52 @@ Go to [Solution](#s3)   [Top](#top)
 <!-- Don't remove -->
 <a name="s1"/>
 
-### 1. SOLUTION 1 TODO :bug:
+### 1. Validate IP Address
 
-Source: TODO :bug:
+Source: [Leetcode](https://leetcode.com/problems/validate-ip-address/)
 
-#### Naive/Simple Solution
+#### Solution
+Python is an excellent language for string manipulation owing to its powerful and intuitive built-in string library. 
 
-TODO :bug:
+We can start this problem by calling the `.split('.')` function on our IP string parameter to separate the input by `.` symbols. 
 
-#### Optimal Solution
+From there, we can perform a quick check of the length of the list returned from the `.split()` using the `len()` function (length). 
+If we don't have exactly four elements in our list, we know the input is invalid.
 
-TODO :bug:
+Otherwise, we will traverse through the four values, checking each one for leading zeros using the `[0]` index operator. 
+Then we  can cast each string value to an int to check that the values are between 0 and 255.
 
-#### Testing The Solutions OR Driver For Solution
+If we are able to traverse through all four values without setting off any of these conditions, we know we have a valid IPv4 
+address. 
+```python
+def isValidIP(IP):
+    values = IP.split('.')
+    if len(values) != 4:
+        return False
+    for str_val in values:
+        if len(str_val) > 1 and str_val[0] == '0':
+            return False
+        int_val = int(str_val)
+        if int_val < 0 or int_val > 255:
+            return False
+    return True
+```
 
-TODO :bug:
+#### Driver For Solution
+
+The solution code is [in the repository](https://github.com/UWB-ACM/CTCI/tree/master/2020-01-Winter/2_string_manipulation/validate_IP/validate_IP.py)
```

Comment:
You can do a relative link to the file instead of an absolute link if you want to, e.g.

```markdown
[in the repository](./validate_IP/validate_IP.py)
```

Reply:
😄 thank you!! and ty for the tip, that looks much cleaner

Reply:
Cool! I edited my comment to include a code suggestion which you can commit directly from the UI if you like. Or, if you make your own edits locally, you can ignore the suggestion. :tada:

Reply:
Or, you can ignore it altogether; it's more of a style preference than a standard.